pinot-controller/src/test/java/com/linkedin/pinot/controller/helix/core/realtime/segment/FlushThresholdUpdaterTest.java

```diff
@@ -114,18 +120,355 @@ private String makeFakeSegmentName(int id) {
     return new LLCSegmentName("fakeTable_REALTIME", id, 0, 1234L).getSegmentName();
   }
 
-  static class FakeFlushThresholdUpdater extends DefaultFlushThresholdUpdater {
+  @Test
+  public void testSegmentSizeBasedFlushThresholdUpdate() {
+    String tableName = "aRealtimeTable_REALTIME";
+    SegmentSizeBasedFlushThresholdUpdater segmentSizeBasedFlushThresholdUpdater =
+        new SegmentSizeBasedFlushThresholdUpdater();
+
+    long startOffset_p0 = 0;
+    int seqNum_p0 = 0;
+    int partitionId_0 = 0;
+    double prevRatio;
+    double currentRatio;
+    int expectedNumRows;
+    long segmentSizeBytes;
+    FlushThresholdUpdaterParams params;
 
-    private int _flushSize;
+    // partition:0 segment:0 - new segment
+    LLCRealtimeSegmentZKMetadata segmentMetadata_p0_0 =
+        getNextSegmentMetadata(tableName, startOffset_p0, partitionId_0, seqNum_p0++);
+    params = new FlushThresholdUpdaterParams();
+    segmentSizeBasedFlushThresholdUpdater.updateFlushThreshold(segmentMetadata_p0_0, params);
+    Assert.assertEquals(segmentMetadata_p0_0.getSizeThresholdToFlushSegment(),
+        SegmentSizeBasedFlushThresholdUpdater.INITIAL_ROWS_THRESHOLD);
+    Assert.assertEquals(segmentMetadata_p0_0.getTimeThresholdToFlushSegment(),
+        SegmentSizeBasedFlushThresholdUpdater.MAX_TIME_THRESHOLD);
 
-    public FakeFlushThresholdUpdater(TableConfig realtimeTableConfig, int flushSize) {
-      super(realtimeTableConfig);
-      _flushSize = flushSize;
-    }
+    // partition:0 segment:0 commits with 200M size
+    segmentSizeBytes = 200 * 1024 * 1024;
```

Comment:
use a visible for testing getter to get the constants from FlushThrsholdUpdateManager, and avoid the tests knowing what those values actally are.